client/components/BudgetGrid/index.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component, PropTypes } from 'react';
+
+import './style.scss';
+
+class BudgetGrid extends Component {
+  static propTypes = {
+    data: PropTypes.object
+  }
+  static defaultProps = {
+    data: {}
+  }
+
+  formatAmount = (amount) => {
+    const isNegative = amount < 0;
+    const formatValue = Math.abs(amount).toLocaleString('en-us', { style: 'currency', currency: 'USD' });
+
+    return {
+      text: `${isNegative ? '-' : ''}${formatValue}`,
+      isNegative
+    };
+  }
+
+  renderRow(data) {
```

Comment:
renderRow should be a stateless component instead of factory method. That way it's more in the spirit of React